src/test/java/com/puppycrawl/tools/checkstyle/xpath/iterators/FollowingIteratorTest.java

```diff
@@ -73,4 +73,25 @@ public void testNoSibling() {
                     .isNull();
         }
     }
+
+    @Test
+    public void testSiblingEnumIsClearedOnExhaustion() throws Exception {
+        final NodeInfo startNode = findNode("CLASS_DEF");
+
+        try (FollowingIterator iterator = new FollowingIterator(startNode)) {
+            NodeInfo node = iterator.next();
+            while (node != null) {
+                node = iterator.next();
+            }
+
+            final java.lang.reflect.Field field =
+                FollowingIterator.class.getDeclaredField("siblingEnum");
+            field.setAccessible(true);
+            final Object siblingEnum = field.get(iterator);
```

Comment:
Sorry, no pure unit testing is allowed.
Please cover it by execution of some xpath filter